lib/manageiq/api/common/users.rb

```diff
@@ -0,0 +1,70 @@
+module ManageIQ
+  module API
+    module Common
+      module Users
+        IDENTITY_KEY = 'x-rh-identity'.freeze
+        VALID_KEYS = %w(account_number user username email is_org_admin)
+
+        def self.current
+          Thread.current[:attr_current_user] = validate
+        end
+
+        def self.with_user(other_headers)
+          # Not Implemented
+          raise "Not Implemented"
+        end
+
+        def self.decode(options = nil, key = IDENTITY_KEY)
+          headers = options ? options : ManageIQ::API::Common::Headers.current
+          raise StandardError, "Requires a valid header object" unless headers
+
+          val = headers.instance_variable_get(:@req).stringify_keys
+          JSON.parse(Base64.decode64(val[key]))
+        end
+
+        def self.encode(val)
```

Comment:
@syncrou I don't think this function is needed, whatever 3 scale generates will be based on the IPP we can't have the clients just adding arbitrary keys into the hash

Reply:
@mkanoor - This ended up here partly because of this comment: https://github.com/ManageIQ/topological_inventory-api/pull/104#discussion_r251944320